### ISSUE TITLE
Hot reload failing intermittently for Blazor Web App projects with Docker Compose

### ISSUE THREAD

**Issue by JonHodgins** [microsoft/DockerTools#415]
@NCarlsonMSFT https://github.com/microsoft/DockerTools/issues/322#issuecomment-1934949539

Here's a minimal reproducible example: two identical Blazor Web App projects created from the default template, HotReloadTest and HotReloadTest2.

Hot reload works in HotReloadTest, but does not work in HotReloadTest2. The output window says "Changes were successfully applied." for both.

Repo: https://github.com/JonHodgins/HotReloadTest

![msedge_GqNbJ0kQQV](https://github.com/microsoft/DockerTools/assets/19355502/53f16429-52cc-4702-960e-885b23d2a247)

**Comment by JonHodgins** [comment]
Some more testing results:
- Hot reload always works if running a single docker container (with and without docker compose)
- Hot reload will fail unpredictably using docker compose with multiple containers (using the above two container repo, sometimes both will fail, both will work, or either one will fail)

**Comment by NCarlsonMSFT** [comment]
@JonHodgins a quick update: I was able to repro this when debugging, but not through Ctrl+F5. I've reached out to the debugger team and they are investigating.

**Comment by lvde0** [comment]
For me it fails to work entirely. With Ctrl+F5 the Hot Reload button in the toolbar is even greyed out. Making changes to a razor file confirms "Changes were successfully applied." but nothing becomes visible. Also running within Docker Compose project.

VS: Version 17.9.2

**Comment by NCarlsonMSFT** [comment]
@lvde0 can you please open a new issue with specific repro instructions? I just tested Blazor Web Apps as a single project in a container and with a compose app using Ctrl+F5, and it worked.

**Comment by lvde0** [comment]
@NCarlsonMSFT I think I narrowed it down to: 

`<DockerDevelopmentMode>Regular</DockerDevelopmentMode>`

Creating a default Blazor.Mud Template, Adding Docker & Docker Compose Orchestration Support supports hot reload without problems. But changing the `DockerDevelopmentMode` seems to break it. Browser Link Dashboard shows no active connections.

**Comment by NCarlsonMSFT** [comment]
@lvde0 you're right that `<DockerDevelopmentMode>Regular</DockerDevelopmentMode>` disables hot reload.
I've pinged the debugger team to see if they have any updates.

**Comment by lvde0** [comment]
@NCarlsonMSFT Even in the latest version of VS I still cannot get Hot Reload to work reliably with two docker projects. It seem to work _sometimes_ when running without debugger (Ctrl + F5), but even that is quite flaky and for some components (like dialogs for instance) it doesn't work at all. 
I am assuming there's no further info about this issue from the VS team?

**Comment by NCarlsonMSFT** [comment]
@lvde0 I've pinged the debugger team, no update yet.
Trying myself, Hot Reload is working for me with two Web API projects when I Ctrl+F5. Can you give me any more details on the repro where it doesn't update? For instance, there are some known limitations, such as compiled CSS files.

**Comment by giuseppe-terrasi** [comment]
I also encounter this problem especially when the solution contains a lot of projects. As a temporary workaround, I usually delete the created container or close the solution and delete the .vs folder if this isn't enough. Also, the problem usually happens when I stop the debugger when the program is paused on a breakpoint.

**Comment by NCarlsonMSFT** [comment]
Hot reload for multiple services with F5 should be fixed in the next preview

**Comment by kevingates** [comment]
@NCarlsonMSFT Any tie to a specific release? And is this only for .Net 9?

**Comment by NCarlsonMSFT** [comment]
@kevingates This is expected to work in .NET 8+ This fix was in a partner team's code base so it's tricky for me to trace the version it was introduced in, but i believe it should be fixed in 17.13 (possibly in 17.12). Are you still seeing an issue?

**Comment by lvde0** [comment]
@NCarlsonMSFT I didn't see the issue in a while so it appears to be fixed. Thank you!

**Comment by dbreshears** [comment]
Thanks for the verification.
